src/Buffer.mo

```diff
@@ -146,4 +146,19 @@ module {
     };
   };
 
+  /// Construct a buffer from an array.
+  public func fromArray<T>(arr: [T]): Buffer<T> {
+    let count = arr.size();
+    let buffer = Buffer<T>(count);
+
+    var i = 0;
+    label l loop {
+      if (i >= count) break l;
+      buffer.add(arr[i]);
+      i += 1;
+    };
+    
+    buffer
+  };
```

Comment:
Thanks! Why not simply

```suggestion
  /// Construct a buffer from an array.
  public func fromArray<T>(arr: [T]): Buffer<T> {
    return { count = arr.size(); elems = Array.thaw(arr) }
  }
```

Reply:
Looks definitely better!
But can it create an instance of the Buffer class? Motoko Playground shows error:
![image](https://user-images.githubusercontent.com/9087446/165313959-e4cd3044-fc16-4d3d-b669-cc216aee4ab8.png)

It was intended to be a *static method* like `let buf = Buffer.fromArray<Nat>([1, 2, 3]);`
Probably it will work as an *instance method* like:
```motoko
let buf = Buffer.Buffer<Nat>(0);
buf.fromArray([1, 2, 3]);
```

![image](https://user-images.githubusercontent.com/9087446/165314777-cff4864a-704d-4e89-8b90-e5ffb9700bb0.png)

Not sure we can leave the name `fromArray` in this case, because in other files in base lib `from*` methods are static.

Link to the playground https://m7sm4-2iaaa-aaaab-qabra-cai.raw.ic0.app/?tag=2337373062

Reply:
Ah, right, Motoko classes have only one constructor usually…

Maybe first add `buffer.addArray` (which can be implemented efficiently using `Array.thraw`, especially if the buffer is empty before), and then the `fromArray` function.